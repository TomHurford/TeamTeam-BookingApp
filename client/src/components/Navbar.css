

.navbar{
    background-color: purple;
    display: flex;
    margin: 0;
    padding: 0;
    
}

ul{
    display: flex;
    margin: 0;
    padding: 0;
    list-style: none;
    justify-content: flex-end;
    width: 100%;
}

ul li{
    color: yellow;
    font-size: 20px;
    font-weight: bold;
    float: right;
    margin: 15px;
}

ul li:hover{
    color: orangered;
}


