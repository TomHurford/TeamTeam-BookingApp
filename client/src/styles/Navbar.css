/*Styling for the navbar and logo below*/

.navbar{
    width: 100%;
    height: 60px;
    background-color: #000000;
    position: fixed;
    top: 0;
    min-width: 700px;
}

.navbarElements{
    display: flex;
    margin: 0;
    padding: 0;
    list-style: none;
    justify-content: flex-end;
    width: 100%;
}

.navbar ul li{
    list-style: none;
    color: white;
    font-size: 25px;
    font-weight: bold;
    float: right;
    margin: 15px;
    height: 60px;
    
}

.navbar ul li:hover{
    cursor: pointer;
    opacity: 0.8;
    transform: scale(1.05);
    transition: 0.3s;
}

.navbar ul li:active{
    transform: scale(1.02);
    transition: 0.3s;
}

.navbar ul li > img{
    width: 30px;
    height: 30px;
}
.navbar ul li.left {
    position: absolute;
    left: 0;
}