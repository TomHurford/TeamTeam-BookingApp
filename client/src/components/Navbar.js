import React, {useState} from 'react';
import '../styles/Navbar.css';
import {Link} from 'react-router-dom';
import logo from '../utils/logo.png';
import basket from '../utils/basket.png';


//Create a navbar component

function Navbar(props) {

    const [notMenuClass, setNotMenuClass] = useState("menu");
    const [menuActive, setMenuActive] = useState("");
    const [menuClass, setMenuClass] = useState('');

    const toggleMenu = () => {
        menuActive ? setMenuClass("") : setMenuClass("menu");
        menuActive ? setNotMenuClass("dead") : setNotMenuClass("");
        setMenuActive(!menuActive);
    }
    const closeMenu = () => {
        setMenuClass("");
        setNotMenuClass("dead");
        setMenuActive(!menuActive);
    }
    const openMenu = () => {
        setMenuClass("menu");
        setNotMenuClass("");
        setMenuActive(!menuActive);
    }

    return (
        <div className='nav-container'>
            <nav className = "navbar">
                <ul className='left'>
                    <Link to = "/"><li className='left'><img src = {logo} alt = "Logo"></img></li></Link>
                    <Link to = "/"><li>Events</li></Link>
                    <Link to = "/societies"><li>Societies</li></Link>
                    <Link to = "/contact"><li>Contact</li></Link>
                </ul>
                { props.isLoggedIn ? 
                <ul className='right'>
                    <Link to = "/basket"><li><img src = {basket} alt = "Basket"></img></li></Link>
                    <Link to = "/logout"><li>Logout</li></Link>
                    <Link to = "/tickets"><li>Tickets</li></Link>
                    <Link to = "/create-event"><li>Create Event</li></Link>
                </ul> : 
                <ul className='right'>
                    <Link to = "/basket"><li><img src = {basket} alt = "Basket"></img></li></Link>
                    <Link to = "/login"><li>Login / SignUp</li></Link>
                </ul> }
            </nav>
            <nav className = "mobile-navbar">
                <ul className='left'>
                    <Link to = "/"><li className='left'><img src = {logo} alt = "Logo"></img></li></Link>
                </ul>
                <ul className='right'>
                    <a onClick={() => {openMenu()}}><li><img src = {basket} alt = "Basket"></img></li></a>
                </ul>
                <div id='notMenu' className={notMenuClass} onClick={() => {closeMenu()}}></div>
                <div id='menu' className={menuClass}>
                    <ul className='top'>                    
                        <a onClick={() => {toggleMenu()}}><li>Close</li></a>
                        <Link onClick={() => {toggleMenu()}} to = "/"><li>Events</li></Link>
                        <Link onClick={() => {toggleMenu()}} to = "/societies"><li>Societies</li></Link>
                        <Link onClick={() => {toggleMenu()}} to = "/contact"><li>Contact</li></Link>
                    </ul>
                    <div className='menuSpacing'></div>
                    { props.isLoggedIn ? 
                    <ul className='bottom'>
                        <Link onClick={() => {toggleMenu()}} to = "/tickets"><li>Tickets</li></Link>
                        <Link onClick={() => {toggleMenu()}} to = "/basket"><li>Basket</li></Link>
                        <Link onClick={() => {toggleMenu()}} to = "/logout"><li>Logout</li></Link>
                    </ul> : 
                    <ul className='bottom'>
                        <Link onClick={() => {toggleMenu()}} to = "/login"><li>Login / SignUp</li></Link>
                        <Link onClick={() => {toggleMenu()}} to = "/basket"><li>Basket</li></Link>
                    </ul> }
                </div>
            </nav>
        </div>
    )
}
export default Navbar
