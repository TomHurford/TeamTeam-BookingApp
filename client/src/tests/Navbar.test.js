import React from "react";
import Navbar from "../components/Navbar";
import { BrowserRouter } from "react-router-dom";
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

test('Navbar snapshot test', () => {
  const component = renderer.create(
    <BrowserRouter>
      <Navbar/>
    </BrowserRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});