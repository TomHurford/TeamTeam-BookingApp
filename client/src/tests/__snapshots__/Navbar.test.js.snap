// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot test 1`] = `
<div
  className="nav-container"
>
  <nav
    className="navbar"
  >
    <ul
      className="left"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <li
          className="left"
        >
          <img
            alt="Logo"
            src="logo.png"
          />
        </li>
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        <li>
          Events
        </li>
      </a>
      <a
        href="/societies"
        onClick={[Function]}
      >
        <li>
          Societies
        </li>
      </a>
      <a
        href="/contact"
        onClick={[Function]}
      >
        <li>
          Contact
        </li>
      </a>
    </ul>
    <ul
      className="right"
    >
      <a
        href="/basket"
        onClick={[Function]}
      >
        <li>
          <img
            alt="Basket"
            src="basket.png"
          />
        </li>
      </a>
      <a
        href="/login"
        onClick={[Function]}
      >
        <li>
          Login / SignUp
        </li>
      </a>
    </ul>
  </nav>
  <nav
    className="mobile-navbar"
  >
    <ul
      className="left"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <li
          className="left"
        >
          <img
            alt="Logo"
            src="logo.png"
          />
        </li>
      </a>
    </ul>
    <ul
      className="right"
    >
      <a
        onClick={[Function]}
      >
        <li>
          <img
            alt="Basket"
            src="burger.png"
          />
        </li>
      </a>
    </ul>
    <div
      className="menu"
      id="notMenu"
      onClick={[Function]}
    />
    <div
      className=""
      id="menu"
    >
      <ul
        className="top"
      >
        <a
          onClick={[Function]}
        >
          <li>
            Close
          </li>
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          <li>
            Events
          </li>
        </a>
        <a
          href="/societies"
          onClick={[Function]}
        >
          <li>
            Societies
          </li>
        </a>
        <a
          href="/contact"
          onClick={[Function]}
        >
          <li>
            Contact
          </li>
        </a>
      </ul>
      <div
        className="menuSpacing"
      />
      <ul
        className="bottom"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          <li>
            Login / SignUp
          </li>
        </a>
        <a
          href="/basket"
          onClick={[Function]}
        >
          <li>
            Basket
          </li>
        </a>
      </ul>
    </div>
  </nav>
</div>
`;
